### PR TITLE
various formulae: simplify `install` methods

### PR DESCRIPTION
**Pull request by carlocab**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

This improves build reproducibility. See #80607.

-----

Needs https://github.com/Homebrew/brew/pull/11659 to lang in a release tag.

### CODE REVIEW

File: Formula/gcal.rb [hunk 1]
```diff
@@ -25,7 +25,7 @@ def install
   end
 
   test do
-    date = Time.now.year
+    date = Time.at(ENV["SOURCE_DATE_EPOCH"].to_i).year
```
Comment (carlocab) [inline Formula/gcal.rb:28]: This one probably won't make a difference, but may as well make it uniform.

Comment (carlocab) [reply Formula/gcal.rb]: Oh, wait, no, we don't want this.

File: Formula/kubebuilder.rb [hunk 2]
```diff
@@ -25,7 +25,7 @@ def install
       -X main.goos=#{goos}
       -X main.goarch=#{goarch}
       -X main.gitCommit=#{Utils.git_head}
-      -X main.buildDate=#{Time.now.iso8601}
+      -X main.buildDate=#{Time.at(ENV["SOURCE_DATE_EPOCH"].to_i).iso8601}
```
Comment (nandahkrishna) [inline Formula/kubebuilder.rb:28]: Is it important to use `utc` everywhere? The timezone isn't set during the build, and I wonder if not using `utc` would be a problem (wrt reproducibility).

**Comment by SMillerDev** [comment]
This looks like it would be worth a formulae DSL. Doesn't have to stop this PR though.

File: Formula/kubebuilder.rb (hunk 2 above)
Comment (carlocab) [reply Formula/kubebuilder.rb:28]: Yea, let's do `utc`.

**Comment by carlocab** [comment]
Homebrew/brew#11659

**Review by nandahkrishna — COMMENTED** [review]
Minor issue with `chezmoi` – need to match `Homebrew` rather than `homebrew` in the test 😅.

**Review by nandahkrishna — DISMISSED** [review]
Thanks, @carlocab! Not removing the `do not merge` label here (since we need to wait for a new Homebrew/brew release).

(I pushed a change for `chezmoi` shortly before you did 🙈 but yours is better.)

**Comment by bayandin** [comment]
I've replaced `Date.today` in the same way.
Probably we could add an audit for this case in the future.

**Comment by nandahkrishna** [comment]
@bayandin I don't know if this would cause problems but the output isn't the same:
```
irb(main):001:0> Date.today.to_s
=> "2021-07-08"
irb(main):002:0> :awsweeper.f.time.iso8601
=> "2021-07-08T16:15:54Z"
```

**Comment by bayandin** [comment]
> @bayandin I don't know if this would cause problems but the output isn't the same:
> 
> ```
> irb(main):001:0> Date.today.to_s
> => "2021-07-08"
> irb(main):002:0> :awsweeper.f.time.iso8601
> => "2021-07-08T16:15:54Z"
> ```

Yep, I don't think so, but changed it to the previous format just in case (except `mage`, because in build script they use `rfc3339`)


And replaced all the usage of `Time.now` to `time`.

**Comment by nandahkrishna** [comment]
Seems like this was cancelled automatically because the `CI-long-timeout` label wasn't found, and the default is 60 minutes. 😅

File: Formula/gdu.rb [hunk 3]
```diff
@@ -18,8 +18,6 @@ class Gdu < Formula
   conflicts_with "coreutils", because: "both install `gdu` binaries"
 
   def install
-    ENV["TZ"] = "UTC"
```
Comment (nandahkrishna) [inline Formula/gdu.rb:21]: Took the opportunity to remove this line, wasn't necessary since the `time` method takes care of UTC.

**Comment by nandahkrishna** [comment]
Default is to fail fast when one matrix runner fails, so CI was cancelled after Linux failed:
```
brew test --retry --verbose localstack
brew test --retry --verbose heartbeat
brew install --build-from-source packetbeat
```
Note: None of these are bottled in homebrew-core yet.

**Review by nandahkrishna — APPROVED** [review]

**Review by chenrui333 — APPROVED** [review]

**Comment by carlocab** [comment]
This should be merged without bottles -- some of the formulae modified here will have had version bumps by the time this PR is merged (e.g. `gdu` in #80910) and we don't want those bottles to be replaced by the ones built here.

**Comment by nandahkrishna** [comment]
We can use `CI-no-bottles` (new label which'll work with scheduled merges).

**Comment by nandahkrishna** [comment]
We can merge this now, Homebrew 3.2.2 was released.

**Pull request closed without merging**